From 51745be13c2454ab5a996e729be7074cd3c26dd0 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 18 Jul 2017 15:28:53 +0800
Subject: [PATCH 2156/5242] MLK-16015-7 gpu: imx: dpu: common: Make
 dpu_pixel_link_config() return void

commit  672a385dbcc077b11defb4d917aae20185c0c678 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch makes dpu_pixel_link_config() return void, since no one is
checking the return value and actually even if any of the SCU operations
inside this function fails we can do nothing about it.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 8b41222..02bee65 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -1176,7 +1176,7 @@ static void dpu_debug_ip_identity(struct dpu_soc *dpu)
 }
 
 /* FIXME: set pixel link in a proper manner */
-static int dpu_pixel_link_config(int id)
+static void dpu_pixel_link_config(int id)
 {
 	sc_err_t sciErr;
 	sc_ipc_t ipcHndl = 0;
@@ -1185,13 +1185,13 @@ static int dpu_pixel_link_config(int id)
 	sciErr = sc_ipc_getMuID(&mu_id);
 	if (sciErr != SC_ERR_NONE) {
 		pr_err("Cannot obtain MU ID\n");
-		return sciErr;
+		return;
 	}
 
 	sciErr = sc_ipc_open(&ipcHndl, mu_id);
 	if (sciErr != SC_ERR_NONE) {
 		pr_err("sc_ipc_open failed! (sciError = %d)\n", sciErr);
-		return sciErr;
+		return;
 	}
 
 	if (id == 0) {
@@ -1267,7 +1267,6 @@ static int dpu_pixel_link_config(int id)
 	}
 
 	sc_ipc_close(mu_id);
-	return sciErr;
 }
 
 static int dpu_probe(struct platform_device *pdev)
-- 
1.7.9.5

